tests/foreman/api/test_ansible.py

```diff
@@ -123,3 +123,79 @@ def test_positive_ansible_job_on_host(target_sat, module_org, rhel_contenthost):
     )
     result = target_sat.api.JobInvocation(id=job['id']).read()
     assert result.succeeded == 1
+
+
+@pytest.mark.no_containers
+def test_positive_ansible_job_on_multiple_host(
+    target_sat,
+    module_org,
+    rhel9_contenthost,
+    rhel8_contenthost,
+    rhel7_contenthost,
+    module_location,
+    module_ak_with_synced_repo,
+):
+    """Test execution of Ansible job on multiple hosts simultaneously.
+
+    :id: 9369feef-466c-40d3-9d0d-65520d7f21ef
+
+    :customerscenario: true
+
+    :steps:
+        1. Register multiple content hosts with satellite
+        2. Import a role into satellite
+        3. Assign that role to all host
+        4. Trigger ansible job keeping all host in a single query
+        5. Check the passing and failing of individual hosts
+        6. Check if one of the job on a host is failed resulting into whole job is marked as failed.
+
+    :expectedresults:
+        1. One of the jobs failing on a single host must impact the overall result as failed.
+
+    :BZ: 2167396, 2190464, 2184117
+
+    :CaseAutomation: Automated
+    """
+    hosts = [rhel9_contenthost, rhel8_contenthost, rhel7_contenthost]
+    SELECTED_ROLE = 'RedHatInsights.insights-client'
+    for host in hosts:
+        command = target_sat.api.RegistrationCommand(
+            organization=module_org,
+            activation_keys=[module_ak_with_synced_repo.name],
+            location=module_location,
+            force=True,
+        ).create()
+        result = host.execute(command)
+        assert result.status == 0
```

Comment:
I would suggest against crafting the command in the test. Use the existing method instead.

What is the reason for forcing the registration?

```suggestion
       host.register(module_org, module_location, module_ak_with_synced_repo.name, target_sat)
       assert host.registered
```